spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -691,12 +691,18 @@ private void initPartitionsIfNeeded() {
 
 					if (offset < 0) {
 						this.consumer.seekToEnd(topicPartition);
-						newOffset = this.consumer.position(topicPartition) + offset;
+						newOffset = Math.max(0, this.consumer.position(topicPartition) + offset);
 					}
 
-					this.consumer.seek(topicPartition, newOffset);
-					if (this.logger.isDebugEnabled()) {
-						this.logger.debug("Reset " + topicPartition + " to offset " + newOffset);
+					try {
+						this.consumer.seek(topicPartition, newOffset);
+						if (this.logger.isDebugEnabled()) {
+							this.logger.debug("Reset " + topicPartition + " to offset " + newOffset);
+						}
+					}
+					catch (Exception e) {
+						logger.error("Failed to set initial offset for " + topicPartition
+								+ " at " + newOffset, e);
```

Comment:
Looks like we can't extract variable so easy:
```
this.consumer.seekToEnd(topicPartition);
newOffset = Math.max(0, this.consumer.position(topicPartition) + offset);
```
That `position` definitely depends on the previous `seekToEnd`.
So, I think if we fail with consequent ``seek`, we should just call `this.consumer.position(topicPartition)` one more time in this `logger.error()` message.

Am I missing anything?